src/main/java/bwapi/BWClient.java

```diff
@@ -7,58 +7,112 @@
  */
 public class BWClient {
     private final BWEventListener eventListener;
-    private final boolean debugConnection;
-    private EventHandler handler;
+    private BotWrapper botWrapper;
+    private Client client;
+    private PerformanceMetrics performanceMetrics;
 
     public BWClient(final BWEventListener eventListener) {
-        this(eventListener, false);
-    }
-
-    /**
-     * @param debugConnection set to `true` for more explicit error messages (might spam the terminal).
-     *                        `false` by default
-     */
-    public BWClient(final BWEventListener eventListener, final boolean debugConnection) {
         Objects.requireNonNull(eventListener);
-        this.debugConnection = debugConnection;
         this.eventListener = eventListener;
     }
 
     /**
      * Get the {@link Game} instance of the currently running game.
+     * When running in asynchronous mode, this is the game from the bot's perspective, eg. potentially a previous frame.
      */
     public Game getGame() {
-        return handler == null ? null : handler.getGame();
+        return botWrapper == null ? null : botWrapper.getGame();
+    }
+
+    /**
+     * @return JBWAPI performance metrics.
+     */
+    public PerformanceMetrics getPerformanceMetrics() {
+        return performanceMetrics;
+    }
+
+    /**
+     * @return The number of frames between the one exposed to the bot and the most recent received by JBWAPI.
+     * This tracks the size of the frame buffer except when the game is paused (which results in multiple frames arriving with the same count).
+     */
+    public int framesBehind() {
```

Comment:
Iis this tested in the synchro test? impl looks good to me, but just making sure

Reply:
Yes, there are a few tests which verify its values.